src/unitycatalog/ai/databricks.py

```diff
@@ -408,17 +410,29 @@ def _execute_uc_function(
             if response.status and job_pending(response.status.state) and response.statement_id:
                 statement_id = response.statement_id
                 _logger.info("Retrying to get statement execution status...")
-                max_retries = 6
-                for retry_cnt in range(1, max_retries + 1):
-                    time.sleep(2**retry_cnt)
+                wait_time = 0
+                retry_cnt = 0
+                client_execution_timeout = int(
+                    os.environ.get(
+                        UNITYCATALOG_AI_CLIENT_EXECUTION_TIMEOUT,
+                        DEFAULT_UC_AI_CLIENT_EXECUTION_TIMEOUT,
+                    )
+                )
+                while wait_time < client_execution_timeout:
+                    wait = min(2**retry_cnt, client_execution_timeout - wait_time)
```

Comment:
any chance that `client_execution_timeout - wait_time` becomes negative?

Reply:
never mind, it's always positive

Reply:
The loop requirement is wait_time < client_execution_timeout